buildSrc/src/main/java/com/google/firebase/gradle/plugins/PackageTransform.kt

```diff
@@ -0,0 +1,321 @@
+  @get:Input abstract val groupId: Property<String>
+  @get:Input abstract val projectPath: Property<String>
+
+  fun getSymbol(line: String): String {
+    val parts = line.split(" ")
+    if (parts.get(0) == "val" || parts.get(0) == "class") {
+      // Field or class
+      return parts.get(1).replace(":", "")
+    } else if (parts.get(0) == "object") {
+      return parts.get(1)
+    } else if (parts.get(0) == "fun") {
+      // Method
+      var output: String = ""
+      var ignore = false
+      for (c in line) {
+        if (c == ':') {
+          ignore = true
+        }
+        if (c == ')' || c == ',') {
+          ignore = false
+        }
+        if (c == '=') {
+          break
+        }
+        if (!ignore) {
+          output += c
+        }
+      }
+      return output.replace("fun", "")
+    }
+    return ""
+  }
+
+  fun copyDir(src: Path, dest: Path) {
+    Files.walk(src).forEach {
+      if (!Files.isDirectory(it)) {
+        val destination = dest.resolve(src.relativize(it))
+        Files.createDirectories(destination.parent)
+        Files.copy(it, destination, StandardCopyOption.REPLACE_EXISTING)
+      }
+    }
+  }
+
+  fun deprecateKTX(src: String, pkgName: String) {
+    File(src).walk().forEach {
+      if (it.absolutePath.endsWith(".kt")) {
+        val lines = File(it.absolutePath).readLines()
+        val output = mutableListOf<String>()
+        for (i in 0 until lines.size) {
+          output.add(lines[i])
+          if (lines[i].contains("*/")) {
+            var symbol = ""
+            var ctr = i + 1
+            while (symbol.isEmpty()) {
+              symbol = getSymbol(lines[ctr++]).trim()
+            }
+            output.add(
+              """@Deprecated("Use `${pkgName}${symbol}`", ReplaceWith("${pkgName}${symbol}"))"""
+            )
+          }
+        }
+        File(it.absolutePath).writeText(output.joinToString("\n"))
+      }
+    }
+  }
+  fun updateKTXReferences(src: String) {
+    // Remove all .ktx suffixes essentially.
+    File(src).walk().forEach {
+      if (it.absolutePath.endsWith(".kt") && !it.absolutePath.contains("/ktx/")) {
+        val lines = File(it.absolutePath).readLines().map { x -> x.replace(".ktx", "") }
+        File(it.absolutePath).delete()
+        val newFile = File(it.absolutePath)
+        newFile.writeText(lines.joinToString("\n"))
+      }
+    }
+  }
+
+  fun updatePlatformLogging(src: String) {
+    File(src).walk().forEach {
+      if (it.absolutePath.endsWith(".kt")) {
+        val lines =
+          File(it.absolutePath).readLines().filter { x ->
+            !x.contains("contains(LIBRARY_NAME)") &&
+              !x.contains("LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME)")
+          }
+        File(it.absolutePath).delete()
+        val newFile = File(it.absolutePath)
+        newFile.writeText(lines.joinToString("\n"))
+      }
+    }
+  }
+
+  fun readDependencies(path: String): List<String> {
```

Comment:
you can read the whole file in a string, then find `dependencies {` and then start processing it.

Reply:
Feel free to ignore since it's not going to be merged to master.